### PR TITLE
Write benchmarks and optimize RequestTemplate::to_request.

### PR DESCRIPTION
**Pull request by shahzad84**
hii check this pull request .  /claim #348

### CODE REVIEW

**Comment by tusharmath** [comment]
@shahzad84 The idea is to write a benchmark and optimize the implementation. You could post before and after results that are produced for the benchmark

**Comment by shahzad84** [comment]
check this commit   2682a0b5d3b0e64524af5440c6566029910a2864

File: Cargo.toml [hunk 1]
```diff
@@ -54,6 +54,8 @@ httpmock = "0.6"
 mockito = "1.2.0"
 pretty_assertions = "1.4.0"
 stripmargin = "0.1.1"
+reqwest = "0.11"
+serde_json = "1.0"
```
Comment (tusharmath) [inline Cargo.toml:58]: They are already dependencies of the project.

**Comment by tusharmath** [comment]
Hey @shahzad84 are you still on this?

File: Cargo.toml (hunk 1 above)
Comment (shahzad84) [reply Cargo.toml]: i have updated the dependencies check it once thank you

**Comment by tusharmath** [comment]
Closing for now. Incase you feel like coming back to this PR sometime later, feel free to reopen with optimizations.

**Pull request closed without merging**